sdk/core/Azure.Core/src/Internal/AzureBaseBuffersExtensions.cs

```diff
@@ -27,12 +27,8 @@ public static async Task WriteAsync(this Stream stream, ReadOnlyMemory<byte> buf
                 }
                 else
                 {
-                    if (array == null || buffer.Length < buffer.Length)
-                    {
-                        if (array != null)
-                            ArrayPool<byte>.Shared.Return(array);
-                        array = ArrayPool<byte>.Shared.Rent(buffer.Length);
-                    }
+                    array = ArrayPool<byte>.Shared.Rent(buffer.Length);
+
                     if (!buffer.TryCopyTo(array))
                         throw new Exception("could not rent large enough buffer.");
```

Comment:
I know this is not part of the PR, but can ArrayPool.Rent return buffer shorter than requested? If no, then this exception is not good. If yes, then instead throwing, we should either allocate a buffer on the heap or borrow a shorter buffer and copy the data in chunks.

Reply:
AFAIK, it can't be.

Reply:
https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs#L144

Reply:
hopefully this will never change.

Reply:
In this case, the exception will probably not going to be thrown. In the sequence case there was another bug that would have led to the exception thrown (because of a wrong check).